packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -52,6 +53,96 @@ async function offerToCreateTemplate(): Promise<void> {
   }
 }
 
+function findFoamVariables(templateText: string): string[] {
+  const regex = /\$(FOAM_[_a-zA-Z0-9]*)|\${(FOAM_[[_a-zA-Z0-9]*)}/g;
+  var matches = [];
+  const output: string[] = [];
+  while ((matches = regex.exec(templateText))) {
+    output.push(matches[1] || matches[2]);
+  }
+  const uniqVariables = [...new Set(output)];
+  return uniqVariables;
+}
+
+function resolveFoamTitle(givenValues: Map<string, string>) {
+  return window.showInputBox({
+    prompt: `Enter a title for the new note`,
+    value: 'Title of my New Note',
+    validateInput: value =>
+      value.trim().length === 0 ? 'Please enter a title' : undefined,
+  });
+}
+
+async function resolveFoamTitleSlug(
+  resolver: Resolver,
+  givenValues: Map<string, string>
+) {
+  const slugger = new GithubSlugger();
+  const foamTitle = await resolver.resolve('FOAM_TITLE', givenValues);
+  return Promise.resolve(slugger.slug(foamTitle));
+}
+
+class Resolver {
+  promises = new Map<string, Thenable<string>>();
+
+  resolve(name: string, givenValues: Map<string, string>): Thenable<string> {
+    if (givenValues.has(name)) {
+      this.promises.set(name, Promise.resolve(givenValues.get(name)));
+    } else if (!this.promises.has(name)) {
+      switch (name) {
+        case 'FOAM_TITLE': {
```

Comment:
minor: throughout the codebase our style for `switch` is to only have `{` at the `switch` level, and not for each individual `case` (TIL that's possible).
I am surprised the linter didn't pick this up 🤔

Reply:
Good catch. Fixed.